File: C++/题目1185：特殊排序.cpp

// 题目1185：特殊排序

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:54
 * @url:http://ac.jobdu.com/problem.php?pid=1185
 * 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/17225139
 */

#include <stdio.h>
#include <algorithm>

using namespace std;
const int maxn = 1002;
int i, n;
int array[maxn];

int main() {
    while (scanf("%d", &n) != EOF) {

        for (i = 0; i < n; i++) {
            scanf("%d", &array[i]);
        }
        if (n == 1) {
            printf("%d\n", array[0]);
            printf("%d\n", -1);
            continue;
        }
        sort(array, array + n);
        printf("%d\n", array[n - 1]);
        for (i = 0; i < n - 2; i++) {
            printf("%d ", array[i]);
        }
        printf("%d\n", array[n - 2]);

    }
    return 0;
}
/**************************************************************
Problem: 1185
User: wangzhenqing
Language: C++
Result: Accepted
Time:30 ms
Memory:1024 kb
****************************************************************/


                        

File: Java/题目1185：特殊排序.java

// 题目1185：特殊排序

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:54
 * @url:http://ac.jobdu.com/problem.php?pid=1185
 * 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/17225139
 */

import java.io.StreamTokenizer;
import java.util.Arrays;

/**
 * @ClassName: Main1046
 * @Description: TODO
 * @author wangzq
 * @date 2013-8-7 下午04:00:39
 * 
 * @version 3.0.0
 */
public class Main {
    public static void main(String[] args) throws Exception {
        StreamTokenizer st = new StreamTokenizer(System.in);
        while (st.nextToken() != StreamTokenizer.TT_EOF) {
            int N = (int) st.nval;
            long[] array = new long[N];
            for (int i = 0; i < N; i++) {
                st.nextToken();
                array[i] = (long) st.nval;
            }
            if (N == 1) {
                System.out.println(array[0]);
                System.out.println(-1);
            } else {
                Arrays.sort(array);
                StringBuffer sb = new StringBuffer();
                sb.append(array[N - 1] + "\n");
                for (int i = 0; i < N - 2; i++) {
                    sb.append(array[i] + " ");
                }
                sb.append(array[N - 2]);
                System.out.println(sb.toString());
            }
        }
    }
}

/**************************************************************
 Problem: 1185
 User: wzqwsrf
 Language: Java
 Result: Accepted
 Time:300 ms
 Memory:31956 kb
 ****************************************************************/




